src/special_arrays.jl

```diff
@@ -129,3 +129,63 @@ function inverse!(x::RealVector, t::CorrCholeskyFactor, U::UpperTriangular)
     end
     x
 end
+
+
+
+"""
+    PosDefCholeskyFactor(n)
+
+Cholesky factor of a symmetric positive-definite matrix of size `n`.
+
+Transforms ``n×(n+1)/2`` real numbers to an ``n×n`` upper-triangular matrix `Ω`, such that
+`Ω'*Ω` is a positive definite matrix.
+"""
+@calltrans struct PosDefCholeskyFactor <: VectorTransform
+    n::Int
+    function PosDefCholeskyFactor(n)
+        @argcheck n ≥ 1 "Dimension should be positive."
+        new(n)
+    end
+end
+
+dimension(t::PosDefCholeskyFactor) = (t.n*(t.n+1)) ÷ 2
+
+function transform_with(flag::LogJacFlag, t::PosDefCholeskyFactor, x::RealVector)
+    @unpack n = t
+    T = extended_eltype(x)
+    ℓ = logjac_zero(flag, T)
+    U = Matrix{T}(undef, n, n)
+    index = firstindex(x)
+    @inbounds for col in 1:n
+        for row in 1:(col-1)
+            U[row, col] = x[index]
+            index += 1
+        end
+        if flag isa NoLogJac
```

Comment:
You could avoid code duplication here by using `transform_with(flag, ...)`. When `flag::NoLogJac`, it is "summed" automatically as a no-op.

Reply:
`transform_with` doesn't seem to work for scalar transforms with scalar args:
```
julia> TransformVariables.transform_with(TransformVariables.NoLogJac(), asℝ₊, 1.0)
ERROR: MethodError: no method matching transform_with(::TransformVariables.NoLogJac, ::TransformVariables.ShiftedExp{true,Float64}, ::Float64)
Closest candidates are:
  transform_with(::TransformVariables.NoLogJac, ::TransformVariables.ScalarTransform, ::AbstractArray{T,1} where T<:Real) at /Users/simon/.julia/dev/TransformVariables/src/scalar.jl:15
  transform_with(::TransformVariables.LogJac, ::TransformVariables.ScalarTransform, ::AbstractArray{T,1} where T<:Real) at /Users/simon/.julia/dev/TransformVariables/src/scalar.jl:18
  transform_with(::TransformVariables.LogJacFlag, ::UnitVector, ::AbstractArray{T,1} where T<:Real) at /Users/simon/.julia/dev/TransformVariables/src/special_arrays.jl:45
  ...
Stacktrace:
 [1] top-level scope at none:0
```